contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,559 @@
+accessModes:
+- ReadWriteOnce
+resources:
+requests:
+storage: 80Gi 
+```
+
+For dynamic provisioning and the use of the storageClass, the user also specifically defines the intent of the volume by 
+indicating the volumeType as block. The provisioner for this class will validate whether or not it supports block and return
+an error if it does not.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+  storageClassName: local-fast 
+  volumeType: block #proposed API change
+accessModes:
+- ReadWriteOnce
+resources:
+requests:
+storage: 80Gi 
+```
+
+## Persistent Volume API Changes:
+For static provisioning the admin creates the volume and also is intentional about how the volume should be consumed. For backwards
+compatibility, the absence of volumeType will default to volumes work today, which are formatted with a filesystem depending on 
+the plug-in chosen. Recycling will not be a supported reclaim policy. Once the user deletes the claim against a PV, the volume will 
+be scrubbed according to how it was bound. The path value in the local PV definition would be overloaded to define the path of the raw
+block device rather than the fileystem path.
+
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+name: local-raw-pv
+spec:
+  volumeType: block #proposed API change
+capacity:
+storage: 100Gi
+local:
+  path: /dev/xvdc 
+accessModes:
+  - ReadWriteOnce
+persistentVolumeReclaimPolicy: Delete 
+```
+
+## Storage Class API Changes:
+For dynamic provisioning, it is assumed that values pass in the parameter section are opaque, thus the introduction of utilizing
+fsType in the StorageClass can be used by the provisioner to indicate how to create the volume. The proposal for this value is
+defined here:
+https://github.com/kubernetes/kubernetes/pull/45345 
+Therefore, a provisioner could potentially provision a block device and install the filesystem onto it by indicating the volumeType
+as 'block' but the fsType as 'xfs'.
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+name: block-volume
+provisioner: kubernetes.io/local-block-glusterfs
+parameters:
+  volumeType: block #opaque value  / plug-in dependent
+  fsType: xfs
+```
+The provisioner (if applicable) should validate the parameters and return and error if the combination specified is not supported.
+This also allows the use case for leveraging a Storage Class for utilizing pre-defined static volumes. By labeling the Persistent Volumes
+with the Storage Class, volumes can be grouped and used according to how they are defined in the class.
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+name: block-volume
+provisioner: no-provisioning 
+parameters:
+```
+  
+# Use Cases
+
+## UC1: 
+
+DESCRIPTION: An admin wishes to pre-create a series of raw block devices to expose as PVs for consumption. The admin wishes to specify the purpose of these devices by specifying 'block' as the volumeType for the PVs.
```

Comment:
Since you're using local volumes as the example, you can say "local raw block devices" in the description.  Or, if you want it to be generic, and not just local, then maybe also give an example of some other plugins, like gce pd or fc.